pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/table/internal/SymbolTableResolver.java

```diff
@@ -385,14 +386,17 @@ private Void visitSwitch(ASTSwitchLike node, @NonNull ReferenceCtx ctx) {
 
             for (ASTSwitchBranch branch : node.getBranches()) {
                 ASTSwitchLabel label = branch.getLabel();
-                // collect all bindings. Maybe it's illegal to use composite label with bindings, idk
-                BindSet bindings =
-                    label.children(ASTPattern.class)
+                // Collect all bindings. Composite label with bindings would be a compilation error.
+                BindSet bindings = label.children(ASTPattern.class)
                          .reduce(BindSet.EMPTY, (bindSet, pat) -> bindSet.union(bindersOfPattern(pat)));
 
                 // visit guarded patterns in label
                 int pushBindings = pushOnStack(f.localVarSymTable(top(), enclosing(), bindings.getTrueBindings()));
                 setTopSymbolTableAndVisit(label, ctx);
+                bindings = label.children(ASTGuard.class).reduce(BindSet.EMPTY,
+                        (bindSet, pat) -> bindSet.union(bindersOfExpr(pat.getGuard())));
```

Comment:
Same here, as there is at most one guard:
```suggestion
                bindings = bindersOfExpr(label.children(ASTGuard.class).map(ASTGuard::getGuard).first());
```
I think `bindersOfExpr` already returns an empty BindSet when the param is null.